### ISSUE TITLE
Rotary Encoder plugin: Add a drop-down list with EncoderType

### ISSUE THREAD

**Issue by vzahradnik** [davetcc/tcMenu#470]
**Is your feature request related to a problem? Please describe.**
TcMenu defines the following enum for the rotary encoder type.
![image](https://github.com/davetcc/tcMenu/assets/4460766/aa9f81d4-4c29-47ad-a28e-c4e3ca68e85b)

The plugin in TcMenu Designer, however, doesn't list all the options. `HALF_CYCLE` is missing and `FULL_CYCLE` is implicit.

![image](https://github.com/davetcc/tcMenu/assets/4460766/64a74e4f-c97c-47f9-b30f-39c87122afe5)

**Describe the solution you'd like**
Instead of having a checkbox titled Quarter cycle encoder, add a new drop-down list with all encoder type options, defaulting to `FULL_CYCLE`.

**Comment by davetcc** [comment]
Agreed that would be better, I'll look at this, but the issue is that existing users could be quite badly affected by the change as their properties may no longer align, and it would default to the wrong value.

As per offline discussion earlier, I think that we should get the state-machine based encoder working first, and then maybe even create an entirely new plugin item for it. Marking this one as deprecated or similar.

**Comment by vzahradnik** [comment]
Yes, I agree. It is not a blocker because users can set this option in the code. This is what I did when I tested the implementation.
